pkg/utils/has/controller.go

```diff
@@ -699,13 +700,20 @@ func (h *SuiteController) RetriggerComponentPipelineRun(component *appservice.Co
 		// To retrigger simple build PipelineRun we just need to update the initial build annotation
 		// in Component CR
 	} else {
-		component, err := h.GetHasComponent(component.GetName(), component.GetNamespace())
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			component, err := h.GetHasComponent(component.GetName(), component.GetNamespace())
+			if err != nil {
+				return fmt.Errorf("failed to get component %q for PipelineRun %q in %q namespace: %+v", component.GetName(), pr.GetName(), pr.GetNamespace(), err)
```

Comment:
`component.GetName()` - this would throw runtime error, because in this case `component` is `nil`

(I know it was my code, sorry 🤦 )

```suggestion
				return fmt.Errorf("failed to get component for PipelineRun %q in %q namespace: %+v", pr.GetName(), pr.GetNamespace(), err)
```